modules/gsp-namespace-role/iam.tf

```diff
@@ -0,0 +1,40 @@
+data "aws_iam_policy_document" "assume-role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.account_id}:role/${var.cluster_name}_kiam_server"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "namespace-defaults" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sqs:SendMessage",
+      "sqs:ReceiveMessage"
+    ]
+
+    resources = ["arn:aws:sqs:*:${var.account_id}:${var.cluster_name}-sqsqueue-${var.namespace_name}-*"]
+  }
+}
+
+resource "aws_iam_role" "namespace" {
```

Comment:
I have concerns that this single "namespace" role will end up being too overloaded (ie imagine a public facing application needs some s3 bucket access for some assets, but ends up having permission to everything since the s3 access is only available in the mega "namespace" role) ... I think having roles like "{cluster}-{namespace}-sqs" would be better 

... but I guess this can be iterated on later unless we think it will be hard to back out of

Reply:
Hm, okay, I think we can get away with changing it later.

Reply:
Roles should be per (tenant) application, surely? A pod can only assume a single role, and if a pod needs access to multiple things (say s3 and sqs) then a "-sqs" role name no longer makes sense.

Reply:
We could think about going down the route of setting predefined SQS and S3 (and whatever else) IAM policies for each namespace, then letting a namespace configure a list of roles it wants to be set up along with the list of predefined policies it wants to attach to each, i.e.:
namespaces:
```
- name: alexs-namespace
  roles:
  - name: app-1
    policies:
    - sqs
    - s3
  - name: app-2
    policies:
    - sqs
```

This way we don't create a bunch of roles that won't be used (just policies), namespaces can say a particular app/pod only gets access to certain AWS services (it's not an all-or-nothing delegation from the namespace down to the pods), and there is then one role per tenant application.

Reply:
That feels like a step in the right direction...but I've no idea how we'd plumb all that together.

Reply:
But this would also allow us to add in some necessary hooks & magic for istio traffic management.

Reply:
Let's talk about how that would help istio work from 4PM?

Reply:
We've discussed this and will go ahead with my proposal.